### PR TITLE
Chore/improve precision of archunit, add translaiton

### PR DESCRIPTION
**Pull request by sarpsahinalp**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Ares
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context
- **New Features**
	- Enhanced error handling with localized messages across various components.
	- Improved build process with a clean build step before packaging.

- **Bug Fixes**
	- Updated error messages for clarity and internationalization.

- **Tests**
	- Added headless checks for tests and improved error message assertions.
	- Introduced a new test for translation key completeness between language files.

- **Chores**
	- Updated YAML configuration files to include localization classes.
	- Removed unused imports and streamlined logic in security test components.
	- Corrected character encoding issues in localization properties files.

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2

### CODE REVIEW

File: src/main/java/de/tum/cit/ase/ares/api/architecture/java/archunit/JavaArchUnitSecurityTestCase.java [hunk 1]
```diff
@@ -95,7 +96,8 @@ public void executeArchitectureTestCase(JavaClasses classes) {
                 default -> throw new UnsupportedOperationException("Not implemented yet");
             }
         } catch (AssertionError e) {
-            throw new SecurityException("Ares Security Error (Reason: Student-Code; Stage: Execution): Illegal Statement found: " + e.getMessage());
+            assert e.getMessage().split("\n").length > 1;
```
Comment (MarkusPaulsen) [inline src/main/java/de/tum/cit/ase/ares/api/architecture/java/archunit/JavaArchUnitSecurityTestCase.java:99]: Remove that assert

File: src/main/java/de/tum/cit/ase/ares/api/architecture/java/archunit/postcompile/JavaArchitectureTestCaseCollection.java [hunk 2]
```diff
@@ -101,7 +101,8 @@ public static ArchRule noClassesShouldImportForbiddenPackages(Set<String> allowe
                     public boolean test(JavaClass javaClass) {
                         return allowedPackages.stream().noneMatch(allowedPackage -> javaClass.getPackageName().startsWith(allowedPackage));
                     }
-                });
+                })
+                .as("TEST TEST");
```
Comment (MarkusPaulsen) [inline src/main/java/de/tum/cit/ase/ares/api/architecture/java/archunit/postcompile/JavaArchitectureTestCaseCollection.java:105]: To be removed.

File: src/main/java/de/tum/cit/ase/ares/api/securitytest/java/JavaSecurityTestCaseFactoryAndBuilder.java [hunk 3]
```diff
@@ -197,7 +198,7 @@ public JavaSecurityTestCaseFactoryAndBuilder(
      */
     private void createSecurityTestCases() {
         //<editor-fold desc="Delete old rules code">
-        //javaAOPMode.reset();
+        // Reset
```
Comment (MarkusPaulsen) [inline src/main/java/de/tum/cit/ase/ares/api/securitytest/java/JavaSecurityTestCaseFactoryAndBuilder.java:201]: Remove this in total

**Review by MarkusPaulsen — CHANGES_REQUESTED** [review]

**Review by MarkusPaulsen — APPROVED** [review]
Code looks good now.

**Review by az108 — APPROVED** [review]
Code 👍

**Pull request merged**